ts/main.ts

```diff
@@ -1,66 +1,326 @@
+    const catImgLeft: HTMLImageElement = document.createElement('img');
+    catImgLeft.setAttribute(
+      'src',
+      categorySets[gridID].leftCategories[i].imgPath,
+    );
+    imgContainerLeft.appendChild(catImgLeft);
+    $leftCategories[i].appendChild(imgContainerLeft);
+    const catTextLeft = document.createElement('h1');
+    catTextLeft.textContent = categorySets[gridID].leftCategories[i].text;
+    $leftCategories[i].appendChild(catTextLeft);
+  }
+} // function that renders Grid Categories based on what number Grid
+
+renderGridCategories(currentGridId);
+
+const $gridContainer = document.querySelector(
+  '.grid-container',
+) as HTMLDivElement;
+if (!$gridContainer) throw new Error('.grid-container query failed!');
+const $modal = document.querySelector('.modal') as HTMLDivElement;
+if (!$modal) throw new Error('.modal query failed!');
+const $input = document.querySelector('input') as HTMLInputElement;
+if (!$input) throw new Error('input query failed');
+
+$gridContainer.addEventListener('click', function (event: Event) {
+  const element = event.target as HTMLElement;
+  console.log(element.classList);
+  if (
+    element.classList.contains('game-square') &&
+    !element.classList.contains('correct')
+  ) {
+    console.log('game sq!');
+    $modal.classList.remove('hidden');
+    element.classList.add('bg-yellow-500', 'selected-sq');
+  }
+});
+
+$modal.addEventListener('click', function (event: Event) {
+  const element = event.target as HTMLElement;
+  const $selectedSq = document.querySelector('.selected-sq') as HTMLDivElement;
+  if (!$selectedSq) throw new Error('.selected-sq query failed!');
+
+  if (element.tagName !== 'BUTTON' && element.tagName !== 'INPUT') {
+    clearSelectedSqResetModal($selectedSq);
+  }
+});
+
+const $form = document.querySelector('form') as HTMLFormElement;
+if (!$form) throw new Error('form query failed!');
+
+$form.addEventListener('submit', function (event: Event) {
+  event.preventDefault();
+  const $selectedSq = document.querySelector('.selected-sq') as HTMLDivElement;
+  if (!$selectedSq) throw new Error('.selected-sq query failed!');
+  const usrInput: string = $input.value;
+  console.log(usrInput);
+
+  // const rowCatNdxStr: string = $selectedSq
+  //   .getAttribute('id')
+  //   ?.charAt(4) as string;
+  // const rowCatNdx: number = parseInt(rowCatNdxStr);
+  // const rowCatCheckFor: string =
+  //   categorySets[currentGridId].leftCategories[rowCatNdx].checkFor;
+  // const rowCatVal: number | string =
+  //   categorySets[currentGridId].leftCategories[rowCatNdx].val;
+
+  // const colCatNdxStr: string = $selectedSq
+  //   .getAttribute('id')
+  //   ?.charAt(6) as string;
+  // const colCatNdx: number = parseInt(colCatNdxStr);
+  // const colCatCheckFor: string =
+  //   categorySets[currentGridId].topCategories[colCatNdx].checkFor;
+  // const colCatVal: number | string =
+  //   categorySets[currentGridId].topCategories[colCatNdx].val;
```

Comment:
If this is code you removed or moved elsewhere you should delete commented out code.